src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -100,8 +100,19 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			{
 				AlterTableCmd *command = (AlterTableCmd *) lfirst(commandCell);
 
-				if (command->subtype == AT_AddConstraint ||
-					command->subtype == AT_DropConstraint)
+				if (command->subtype == AT_AddConstraint)
+				{
+					Constraint *con = castNode(Constraint, command->def);
+
+					if (con->contype == CONSTR_PRIMARY && con->indexname)
```

Comment:
What information `con->options` include? Do we ever need to check that? It is mentioned as /* options from WITH clause */ in the code. So, what is `WITH` clause?

Reply:
`WITH` is used in postgres to define a couple of variables in indexes such as 
```SQL
CREATE UNIQUE INDEX u_ind3 on t(i) with (fillfactor=50);
```
means the 50% of the index page will be left empty. There are a couple of other options for different kind of indexes. 

When I run some queries such as;

Reply:
> So, there is a problem on this with Citus and there is no available tests for this right now.

Let's open an issue for that and refer to this discussion.